plutus-scb/src/Plutus/SCB/Webserver/Handler.hs

```diff
@@ -0,0 +1,212 @@
+import           Plutus.SCB.Events                               (ChainEvent, ContractInstanceId (ContractInstanceId),
+                                                                  ContractInstanceState (ContractInstanceState),
+                                                                  csContractDefinition)
+import qualified Plutus.SCB.Query                                as Query
+import           Plutus.SCB.Types
+import           Plutus.SCB.Webserver.Types
+import           Servant                                         ((:<|>) ((:<|>)))
+import           Wallet.Effects                                  (ChainIndexEffect, confirmedBlocks)
+import           Wallet.Emulator.Wallet                          (Wallet)
+import qualified Wallet.Rollup                                   as Rollup
+
+healthcheck :: Monad m => m ()
+healthcheck = pure ()
+
+getContractReport ::
+       forall t effs.
+       ( Member (EventLogEffect (ChainEvent t)) effs
+       , Member (ContractEffect t) effs
+       , Ord t
+       )
+    => Eff effs (ContractReport t)
+getContractReport = do
+    installedContracts <-
+        Set.toList <$> runGlobalQuery (Query.installedContractsProjection @t)
+    crAvailableContracts <-
+        traverse
+            (\t -> ContractSignatureResponse t <$> exportSchema t)
+            installedContracts
+    crActiveContractStates <-
+        Map.elems <$> runGlobalQuery (Query.contractState @t)
+    pure ContractReport {crAvailableContracts, crActiveContractStates}
+
+getChainReport ::
+       forall t effs. Member ChainIndexEffect effs
+    => Eff effs (ChainReport t)
+getChainReport = do
+    blocks :: Blockchain <- confirmedBlocks
+    let ChainOverview { chainOverviewBlockchain
+                      , chainOverviewUnspentTxsById
+                      , chainOverviewUtxoIndex
+                      } = mkChainOverview blocks
+    let walletMap :: Map PubKeyHash Wallet = Map.empty -- TODO Will the real walletMap please step forward?
+    annotatedBlockchain <- Rollup.doAnnotateBlockchain chainOverviewBlockchain
+    pure
+        ChainReport
+            { transactionMap = chainOverviewUnspentTxsById
+            , utxoIndex = chainOverviewUtxoIndex
+            , annotatedBlockchain
+            , walletMap
+            }
+
+getEvents ::
+       forall t effs. (Member (EventLogEffect (ChainEvent t)) effs)
+    => Eff effs [ChainEvent t]
+getEvents = fmap streamEventEvent <$> runGlobalQuery Query.pureProjection
+
+getFullReport ::
+       forall t effs.
+       ( Member (EventLogEffect (ChainEvent t)) effs
+       , Member (ContractEffect t) effs
+       , Member ChainIndexEffect effs
+       , Ord t
+       )
+    => Eff effs (FullReport t)
+getFullReport = do
+    events <- fmap streamEventEvent <$> runGlobalQuery Query.pureProjection
+    contractReport <- getContractReport
+    chainReport <- getChainReport
+    pure FullReport {contractReport, chainReport, events}
```

Comment:
isn't it nice to do the record `{ ... }` thing here? I mean you have just about every extension enabled anyway 😛

Reply:
:-D

I think the overall team consensus is `NamedFieldPuns` yes, `RecordWildCards` no. So I'm trying to follow that. :-)